backend/internal/handlers/classrooms/assignments/assignments.go

```diff
@@ -170,39 +176,65 @@ func (s *AssignmentService) useAssignmentToken() fiber.Handler {
 			return errs.InternalServerError()
 		}
 
-		templateRepoName := assignmentWithTemplate.Template.TemplateRepoName
-		templateRepoOwner := assignmentWithTemplate.Template.TemplateRepoOwner
-
-		// Generate fork name
-		forkName := generateForkName(templateRepoName, user.Login)
-
 		// Check if fork already exists
+		forkName := generateForkName(assignment.Name, user.Login)
 		studentWorkRepo, _ := client.GetRepository(c.Context(), classroom.OrgName, forkName)
-		if studentWorkRepo != nil { // Fork already exists, early return
+		if studentWorkRepo != nil {
+			// Ensure student team is removed
+			err = client.RemoveRepoFromTeam(c.Context(), classroom.OrgName, *classroom.StudentTeamName, classroom.OrgName, forkName)
+			if err != nil {
+				return errs.GithubAPIError(err)
+			}
+
 			return c.Status(http.StatusOK).JSON(fiber.Map{
 				"message":  "Assignment already accepted",
 				"repo_url": studentWorkRepo.HTMLURL,
 			})
 		}
 
-		// Generate Fork
-		err = client.ForkRepository(c.Context(), templateRepoOwner, classroom.OrgName, templateRepoName, forkName)
+		// Otherwise generate fork
+		err = client.ForkRepository(c.Context(),
+			baseRepo.BaseRepoOwner,
+			classroom.OrgName,
+			baseRepo.BaseRepoName,
+			forkName)
+		if err != nil {
+			return errs.GithubAPIError(err)
+		}
+
+		// Wait to perform actions on the fork until it is finished initializing
+		initialDelay := 1 * time.Second
+		maxDelay := 30 * time.Second
```

Comment:
There may be a max timeout for http requests. This is making assumptions about how our frontend and backend are designed and deployed. 
Either way, we need to test this by simulating a long delay when forking

Reply:
How would you suggest we do this, then?